generator/uscore/metadata_extractor.rb

```diff
@@ -226,6 +228,30 @@ def add_search_param_descriptions(profile_definition, sequence)
         end
       end
 
+      def add_valid_codes(profile_definition, profile_element, fhir_metadata, param_metadata, path)
+        if param_metadata[:contains_multiple]
+          # look for values in slices if param is an array
+          slices = profile_definition['snapshot']['element'].select { |el| el['path'] == path && el['sliceName'] }
+          slices.each do |slice|
+            param_metadata[:values] << slice['patternCodeableConcept']['coding'].first['code'] if slice['patternCodeableConcept']
+          end
+        elsif param_metadata[:type] == 'CodeableConcept'
+          # look for fixed codes if codeable concept
+          fixed_code_els = profile_definition['snapshot']['element'].select { |el| el['path'] == "#{profile_element['path']}.coding.code" && el['fixedCode'].present? }
+          param_metadata[:values] += fixed_code_els.map { |el| el['fixedCode'] }
+        end
+        param_metadata[:values] << profile_element['patternCodeableConcept']['coding'].first['code'] if profile_element['patternCodeableConcept']
+        # look for values in the valueset the element is bound to
+        valueset_binding = profile_element['binding']
+        if valueset_binding
+          value_set = resources_by_type['ValueSet'].find { |res| res['url'] == valueset_binding['valueSet'] }
+          codes = value_set['compose']['include'].reject { |code| code['concept'].nil? } if value_set.present?
+          param_metadata[:values] += codes.map { |code| code['concept'].first['code'] } if codes.present?
+        end
+        # if none found so far, add all the valid codes from metadata (less strict)
+        param_metadata[:values] = fhir_metadata['valid_codes'].values.flatten unless param_metadata[:values].any? || fhir_metadata.nil? || fhir_metadata['valid_codes'].nil?
```

Comment:
Use `.present?/.blank?` rather than `.any?` or `.nil?`. `.any?` isn't nil-safe, and `.nil?` returns true for empty arrays, hashes, and strings, which usually isn't what you want.

Reply:
Fixed